script.js

```diff
@@ -116,12 +117,52 @@ class Game {
     this.offsetX = offsetX;
     this.offsetY = offsetY;
     this.length = snakeLength;
+    this.keys = [];
     this.field = new Field(this.width, this.height, this.gridSize, this.offsetX, this.offsetY);
     this.snake = new Snake(this.field, this.length);
     this.stats = stats(width, height, gridSize, offsetX, offsetY);
     this.playButton = document.getElementsByClassName('play-button')[0];
 
     this.start = this.start.bind(this);
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+  }
+  
+  handleKeyPress(e) {
+    this.keys.push(e.key);
+  }
+
+  checkKeys() {
+    if (this.keys.length === 0) {
+      return
+    }
+    const key = this.keys.shift();
+    const direction = this.snake.direction;
+    switch (key) {
+      case 'ArrowUp':
+        if (direction == Snake.DOWN) {
+          return
+        }
+        this.snake.direction = Snake.UP;
+        break;
+      case 'ArrowDown':
+        if (direction == Snake.UP) {
+          return
+        }
+        this.snake.direction = Snake.DOWN;
+        break;
+      case 'ArrowRight':
+        if (direction == Snake.LEFT) {
+          return
+        }
+        this.snake.direction = Snake.RIGHT;
+        break;
+      case 'ArrowLeft':
+        if (direction == Snake.RIGHT) {
+          return
+        }
+        this.snake.direction = Snake.LEFT;
+        break;
+    }
```

Comment:
Would it be more readable if you wrote it as a series of `if`/`else` statements?
```suggestion
    if (key === 'ArrowUp' && direction !== Snake.DOWN) {
      this.snake.direction = Snake.UP;
    }
    else if ((key === 'ArrowDown' && direction !== Snake.UP) {
      this.snake.direction = Snake.DOWN;
    }
    else if ((key === 'ArrowLeft' && direction !== Snake.RIGHT) {
      this.snake.direction = Snake.LEFT;
    }
    else if ((key === 'ArrowRight' && direction !== Snake.LEFT) {
      this.snake.direction = Snake.RIGHT;
    }
```